src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -464,6 +471,110 @@ PRIMARY KEY ({primaryKeys})
+        {
+            const int minMetricSamplesForScaling = 5;
+            const int maxChangesPerWorker = 1000;
+
+            var status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            // Do not make a scale decision unless we have enough samples.
+            if (metrics == null || (metrics.Length < minMetricSamplesForScaling))
+            {
+                return status;
+            }
+
+            // Add worker is count of unprocessed changes per worker exceeds the maximum limit.
+            long lastUnprocessedChangeCount = metrics.Last().UnprocessedChangeCount;
+            this._logger.LogInformation($"Unprocessed change count: {lastUnprocessedChangeCount}, worker count: {workerCount}, maximum changes per worker: {maxChangesPerWorker}.");
+
+            if (lastUnprocessedChangeCount > workerCount * maxChangesPerWorker)
+            {
+                status.Vote = ScaleVote.ScaleOut;
+                this._logger.LogInformation($"Requesting scale-out: Found too many unprocessed changes for table: '{this._userTable.FullName}' relative to the number of workers.");
+                return status;
+            }
+
+            // Check to see if there are no changes occurring in the user table for a while.
+            if (metrics.All(p => p.UnprocessedChangeCount == 0L))
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                this._logger.LogInformation($"Requesting scale-in: Found table: '{this._userTable.FullName}' to be idle for a while.");
+                return status;
+            }
+
+            // Check if there is a continuous increase in count of unprocessed changes.
+            if (metrics[0].UnprocessedChangeCount > 0)
+            {
+                bool isIncreasing = true;
+                for (int index = metrics.Length - minMetricSamplesForScaling; index < metrics.Length - 1; index++)
+                {
+                    if (metrics[index].UnprocessedChangeCount >= metrics[index + 1].UnprocessedChangeCount)
+                    {
+                        isIncreasing = false;
+                        break;
+                    }
+                }
+
+                if (isIncreasing)
+                {
+                    status.Vote = ScaleVote.ScaleOut;
+                    this._logger.LogInformation($"Requesting scale-out: Found the unprocessed changes for table: '{this._userTable.FullName}' to be continuously increasing.");
+                    return status;
+                }
+            }
+
+            // Check if there is a continuous decrease in count of unprocessed changes.
+            bool isDecreasing = true;
+            for (int index = metrics.Length - minMetricSamplesForScaling; index < metrics.Length - 1; index++)
+            {
+                if (metrics[index].UnprocessedChangeCount == 0 || metrics[index].UnprocessedChangeCount > metrics[index + 1].UnprocessedChangeCount)
```

Comment:
Why do we not consider it decreasing if one of the metrics has 0 unprocessed changes? If I have metrics like this 

100 -> 75 -> 50 -> 25 -> 0

then we should scale it down since it's clearly decreasing

Reply:
I reversed the condition by mistake. Please ignore it for now. The intention was to recommend a scale-in even for metric samples like [20, 10, 0, 0, 0] where the plain strictly-decreasing condition would return false.